lustre/tests/test_unit.py

```diff
@@ -0,0 +1,475 @@
+            {'name': 'lustre-MDT0001', 'type': 'mdt'},
+            {'name': 'lustre2-MDT0000', 'type': 'mdt'},
+            {'name': 'lustre-OST0000', 'type': 'ost'},
+        ]
+        filesystems = ['lustre', 'lustre2']
+
+        check._update_changelog_targets(devices, filesystems)
+
+        expected_targets = ['lustre-MDT0000', 'lustre-MDT0001', 'lustre2-MDT0000']
+        assert set(check.changelog_targets) == set(expected_targets)
+
+
+def test_lnet_group_filtering(aggregator, instance, mock_lustre_commands):
+    mapping = {
+        'lctl get_param -ny version': 'all_version.txt',
+        'lctl dl': 'client_dl_yaml.txt',
+        'lnetctl net show': 'all_lnet_net.txt',
+    }
+    with mock_lustre_commands(mapping):
+        check = LustreCheck('lustre', {}, [instance])
+
+        # Test that ignored groups are not submitted
+        test_group = {'test_metric': 123}
+        tags = ['test_tag:value']
+
+        check._submit_lnet_metric_group('local', 'statistics', test_group, tags)
+
+        aggregator.assert_metric('lustre.net.local.statistics.test_metric')
+        aggregator.assert_metric_has_tag('lustre.net.local.statistics.test_metric', 'test_tag:value')
+
+
+def test_metric_type_assignment(aggregator, instance, mock_lustre_commands):
+    mapping = {
+        'lctl get_param -ny version': 'all_version.txt',
+        'lctl dl': 'client_dl_yaml.txt',
+    }
+    with mock_lustre_commands(mapping):
+        check = LustreCheck('lustre', {}, [instance])
+
+        # Test gauge metric
+        check._submit('test.gauge', 100, 'gauge', tags=[])
+        aggregator.assert_metric('lustre.test.gauge', value=100, metric_type=aggregator.GAUGE)
+
+        # Test count metric
+        check._submit('test.count', 50, 'count', tags=[])
+        aggregator.assert_metric('lustre.test.count', value=50, metric_type=aggregator.MONOTONIC_COUNT)
+
+        # Test rate metric
+        check._submit('test.rate', 25, 'rate', tags=[])
+        aggregator.assert_metric('lustre.test.rate', value=25, metric_type=aggregator.RATE)
+
+
+def test_empty_command_outputs(instance, mock_lustre_commands):
+    mapping = {
+        'lctl get_param -ny version': 'all_version.txt',
+        'lctl get_param -ny mdt': '',
+        'lctl dl': 'client_dl_yaml.txt',
+    }
+    with mock_lustre_commands(mapping):
+        check = LustreCheck('lustre', {}, [instance])
+
+        # Should handle empty outputs gracefully
+        result = check._parse_stats('')
+        assert result == {}
+
+        result = check._get_jobstats_metrics('mdt.lustre-MDT0000.job_stats')
+        assert result == {}
+
+
+def test_exception_handling(mock_lustre_commands):
+    """Test various exception handling scenarios."""
+    mapping = {
+        'lctl get_param -ny version': 'all_version.txt',
+        'lctl dl': 'client_dl_yaml.txt',
+    }
+
+    with mock_lustre_commands(mapping):
+        check = LustreCheck('lustre', {}, [{}])
+
+        # Test _find_node_type exception handling
```

Comment:
suggestion- you can also use the caplog fixture to test that certain debug logs were emitted in these cases